### ISSUE TITLE
Clustered Scheduler

### ISSUE THREAD

**Issue by rymsha** [enonic/xp#8596]
# Requirements

core must be able to schedule tasks
- audit-log repository cleanup
- vacuum
- layers sync

user must be able to schedule tasks 
 - scheduled named task
 - deferred publishing

core must have predefined schedules, created on first start

 - audit-log cleanup every day
 - vacuum every Sunday
 - layers-sync every 15 minutes

system predefined and developer defined schedules should be configurable

- disable/enable
- cron-like string to determine frequency
- once on specific time (only developer defined, useful for deferred publishing)

scheduled tasks must be cluster-safe

- run only once in defined period in entire cluster (best effort)

# Scope

- SchedulerService API
- core-scheduler module
-  lib-scheduler
- management endpoints
- core predefined jobs

# Out of Scope

- GUI for task management is a separate application
- CLI 
- who can schedule? only system.admin role can modify system.scheduler repository
- schedule from configuration files. configuration files may be different on different XP nodes, so it is ambiguous.
- deferred publishing
